realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -412,7 +427,8 @@ private void completedAsyncRealmResults(QueryUpdateTask.Result result) {
                         // swap pointer
                         realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
                         // notify callbacks
-                        realmResults.notifyChangeListeners();
+                        realmResults.syncIfNeeded();
```

Comment:
Yes, that call also update the cached version inside RealmResults. This is needed to detect if the listeners should be triggered at all.